### ISSUE TITLE
False positive for redundant-keyword-arg in class with partial() class attribute function

### ISSUE THREAD

**Issue by dgilland** [PyCQA/pylint#2271]
### Steps to reproduce

Create file with these contents:

```python
# false_redundant_keyword_arg.py

from functools import partial

def foo(x, y):
    return (x, y)

bar = partial(foo, '')

class A(object):
    bar = bar

    def b(self):
        y = 1
        # False positive error here.
        # E1124: Argument 'y' passed by position and keyword in function call (redundant-keyword-arg)
        self.bar(y=y)

        # No error here
        bar(y=y)

y = 2
# No error here
bar(y=y)
```

### Current behavior

With Python 3.7 and 3.6 with astroid-2.0.0.dev4 and pylint-2.0.0.dev2:

```
$ pylint -E false_redundant_keyword_arg.py
************* Module false_redundant_keyword_arg
false_redundant_keyword_arg.py:15:8: E1124: Argument 'y' passed by position and keyword in function call (redundant-keyword-arg)
```

Behavior isn't present for Python 3.6 with pylint 1.9.2 and astroid 1.6.5.

### Expected behavior

Should not be an error.

### pylint --version output

```
$ pylint --version output
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.7.0 (default, Jul  6 2018, 19:08:39)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-18)]
```

```
$ pylint --version output
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.6.6 (default, Jul  6 2018, 11:59:38)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-18)]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce the bug!

**Comment by eduardomezencio** [comment]
Still present with
```
pylint 2.2.2
astroid 2.1.0
Python 3.7.1 (default, Oct 22 2018, 10:41:28) 
[GCC 8.2.1 20180831]
```

**Comment by PCManticore** [comment]
@eduardomezencio Yes, hasn't been fixed yet.

**Comment by JordanKadish** [comment]
Getting this issue in my code as well...
- Parent class sets default args
- child calls self.func(<args to replace defaults>)
- pylint gives error `Argument <arg name> passed by position and keyword in classmethod call pylint(redundant-keyword-arg)`

**Comment by richmilne** [comment]
Just wanted to record that I recently came across this behaviour myself, in our development environment:

```
$ pylint --version
pylint 2.13.3
astroid 2.15.4
Python 3.7.4 (default, Oct 17 2019, 06:18:21) 
[GCC 6.3.0 20170516]
```

Those aren't the most recent versions, so I repeated the test in the most recent python 3.7 Debian container (`docker image pull python:3.7.16-bullseye`, `sha256:87864cfea7cb694b216e2e385519dad76d9c2dfbf404d05ef38572c51e43c11d`).

```
$ pylint --version
pylint 2.17.4
astroid 2.15.5
Python 3.7.16 (default, May  4 2023, 06:28:25) 
[GCC 10.2.1 20210110]
```
pylint 2.17.4 is currently the most recent (non-pre-release) version available, and this behaviour is still present there.

**Comment by mharding-hpe** [comment]
Seeing this false error reported in our code as well. Going to have to add a pylint disable statement to avoid it.
